extensions/notebook/src/jupyter/jupyterServerInstallation.ts

```diff
@@ -249,25 +258,43 @@ export default class JupyterServerInstallation {
 				operation: op => {
 					this.installDependencies(op)
 						.then(() => {
-							this._installCompleteEmitter.fire();
+							this._installReady.resolve();
 							updateConfig();
 						})
 						.catch(err => {
 							let errorMsg = msgDependenciesInstallationFailed(err);
 							op.updateStatus(azdata.TaskStatus.Failed, errorMsg);
 							this.apiWrapper.showErrorMessage(errorMsg);
-							this._installCompleteEmitter.fire(errorMsg);
+							this._installReady.reject(errorMsg);
 						});
 				}
 			});
 		} else {
 			// Python executable already exists, but the path setting wasn't defined,
 			// so update it here
-			this._installCompleteEmitter.fire();
+			this._installReady.resolve();
 			updateConfig();
 		}
 	}
 
+	public async promptForPythonInstall(): Promise<void> {
+		if (!JupyterServerInstallation.isPythonInstalled(this.apiWrapper)) {
+			let doInstall = await (new CodeAdapter()).promptSingle<boolean>(<IQuestion>{
+				type: QuestionTypes.confirm,
+				message: localize('confirmPythonInstall', 'Python is required for Jupyter Notebooks. Would you like to install it?'),
+				default: true
+			});
+
+			if (doInstall) {
+				let pythonDialog = new ConfigurePythonDialog(this.apiWrapper, this.outputChannel, this);
+				pythonDialog.showDialog();
```

Comment:
Please have this return a promise that we can await, and if it's canceled instead of "OK" being clicked, it rejects. Otherwise we get into the state where we wait forever on the installReady.promise. Thanks!